src/replica/python/schema_migration.py

```diff
@@ -99,21 +98,21 @@ def _set_version(self, version: int) -> None:
         _log.info(f"Set replica schema version to {version}.")
         self.connection.commit()
 
-        # read it back and compare with expected
+        # Read it back and compare with expected.
         current = self.current_version()
         if current != version:
             raise RuntimeError(
-                f"Failed to update version number in database to {version}, current version is now {current}")
+                f"Failed to update version of database {database} to {version}, current version is {current}.")
 
     def _create_database(self) -> None:
         """Create the replication controller database.
         """
         with closing(self.connection.cursor()) as cursor:
-            cursor.execute(f"CREATE DATABASE {self.database};")
+            cursor.execute(f"CREATE DATABASE {database};")
             warnings = cursor.fetchwarnings()
             if warnings:
-                _log.warn("Warnings were creating database %s", self.database)
-        _log.info(f"Created database {self.database}.")
+                _log.warn(f"Warnings were creating database {database}.")
```

Comment:
f strings should not be used in _log statements (the previous way is the correct the way). This is because it's more efficient to delay substituting the parameters into the string as much as possible, in case the string won't be used.

Reply:
Thanks! I didn't know that. I will put the original code back.